internal/vm-console-proxy-bundle/bundle.go

```diff
@@ -36,109 +36,58 @@ type Bundle struct {
 	ConfigMap          core.ConfigMap
 }
 
-func ReadBundle(path string) (Bundle, error) {
-	files, err := readFile(path)
+func ReadBundle(path string) (*Bundle, error) {
+	data, err := os.ReadFile(path)
 	if err != nil {
-		return Bundle{}, err
-	}
-
-	bundleObj, err := decodeObjectsFromFiles(files)
-	if err != nil {
-		return Bundle{}, err
+		return nil, err
 	}
 
-	return *bundleObj, nil
+	return loadBundleFromBytes(data)
 }
 
 func GetBundlePath() string {
 	return filepath.Join(vmConsoleProxyBundleDir, "vm-console-proxy.yaml")
 }
 
-func readFile(fileName string) ([][]byte, error) {
-	file, err := os.ReadFile(fileName)
-	if err != nil {
-		return nil, err
-	}
-
-	return [][]byte{file}, nil
-}
-
-func decodeObjectsFromFiles(files [][]byte) (*Bundle, error) {
+func loadBundleFromBytes(data []byte) (*Bundle, error) {
 	bundle := &Bundle{}
-	for _, file := range files {
-		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(file), 1024)
-		for {
-			var obj map[string]interface{}
-			err := decoder.Decode(&obj)
-			if err == io.EOF {
-				break
-			}
-			if err != nil {
-				return nil, err
-			}
-			if kind, ok := obj["kind"].(string); ok {
-				if kind == "" {
-					continue
-				}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1024)
+	for {
+		obj := &unstructured.Unstructured{}
+		err := decoder.Decode(&obj)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
 
-				switch kind {
-				case serviceAccountKind:
-					serviceAccount := core.ServiceAccount{}
-					err = getObject(obj, &serviceAccount)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ServiceAccount = serviceAccount
-				case clusterRoleKind:
-					clusterRole := rbac.ClusterRole{}
-					err = getObject(obj, &clusterRole)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ClusterRole = clusterRole
-				case clusterRoleBindingsKind:
-					clusterRoleBinding := rbac.ClusterRoleBinding{}
-					err = getObject(obj, &clusterRoleBinding)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ClusterRoleBinding = clusterRoleBinding
-				case serviceKind:
-					service := core.Service{}
-					err = getObject(obj, &service)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Service = service
-				case deploymentKind:
-					deployment := apps.Deployment{}
-					err = getObject(obj, &deployment)
-					if err != nil {
-						return nil, err
-					}
-					bundle.Deployment = deployment
-				case configMapKind:
-					configMap := core.ConfigMap{}
-					err = getObject(obj, &configMap)
-					if err != nil {
-						return nil, err
-					}
-					bundle.ConfigMap = configMap
-				default:
-					return nil, fmt.Errorf("unsupported Kind found in vm-console-proxy bundle: %s", kind)
-				}
-			}
+		kind := obj.GetKind()
```

Comment:
Merge these lines into the switch?

Reply:
Also, could an empty Kind be a decoding error?

Reply:
Done